test_runner/regress/test_bad_connection.py

```diff
@@ -1,10 +1,13 @@
 import random
 import time
 
+import psycopg2.errors
+import pytest
 from fixtures.log_helper import log
 from fixtures.neon_fixtures import NeonEnvBuilder
 
 
+@pytest.mark.timeout(600)
```

Comment:
We don't really want this test to take 10 minutes, right?

If the reason it should take ten minutes is just the number of retries/backoffs, then I'd suggest tuning the test to fit within the  default 5 minute timeout, rather than running for longer.

Reply:
After discussing a bit, it seems fine to leave this at 10 min. If it's a problem we can always change it, but I in general expect a stress test to take longer anyway.